trestle/core/commands/author/profile.py

```diff
@@ -190,17 +206,22 @@ def _replace_alter_adds(profile: prof.Profile, alters: List[prof.Alter]) -> None
             profile.modify.alters = new_alters
 
     @staticmethod
-    def _replace_modify_set_params(profile: prof.Profile, param_dict: Dict[str, str]) -> None:
-        if param_dict:
+    def _replace_modify_set_params(profile: prof.Profile, param_str_dict: Dict[str, str]) -> None:
+        """Update the set_params in the profile with new values from markdown."""
+        if param_str_dict:
             if not profile.modify:
                 profile.modify = prof.Modify()
             profile.modify.set_parameters = as_list(profile.modify.set_parameters)
-            orig_param_dict = {}
+            # create mapping of original param_id to value for profile
+            # then update it with new or additional mappings in param_str_dict
+            profile_param_str_dict = {}
             for set_param in profile.modify.set_parameters:
-                orig_param_dict[set_param.param_id] = ControlIOReader.param_values_as_string(set_param)
-            orig_param_dict.update(param_dict)
+                param = CatalogInterface.setparam_to_param(set_param.param_id, set_param)
+                param_str = ControlIOReader.param_to_str(param, ParameterRep.VALUE_OR_LABEL_OR_CHOICES)
+                profile_param_str_dict[set_param.param_id] = param_str
+            profile_param_str_dict.update(param_str_dict)
```

Comment:
More of a comment just to make sure that the four following scenarios are handled correctly:
1. User deletes some set-parameter from the header
2. User adds new set-parameter for existing parameter to the header
3. User selects invalid value for the parameter
4. User adds new set-parameter for non-existing parameter to the header (say user by mistake tries to set the parameters that does not exists)

Reply:
I have changed the logic of how this works so that the list of parameters in the markdown is 1:1 with the setparameters in the profile.  Without building the resolved catalog I cannot handle 3 or 4 but errors would be reported for 3 if the catalog is ever resolved by the profile.

Reply:
As discussed this makes sense, and I think this is the way it should be